src/cipher.rs

```diff
@@ -1161,6 +1161,23 @@ impl Castable for rustls_server_cert_verifier {
 }
 
 impl rustls_server_cert_verifier {
+    /// Create a new server certificate verifier that uses the system's root store and WebPKI via
+    /// `rustls-platform-verifier`.
+    ///
+    /// The verifier can be used in several `rustls_client_config` instances and must be freed by
+    /// the application when no longer needed.
+    #[no_mangle]
+    pub extern "C" fn rustls_platform_server_cert_verifier(
+        verifier_out: *mut *mut rustls_server_cert_verifier,
```

Comment:
This could also take an (optional) `rustls_root_cert_store` to use for [`Verifier::new_with_extra_roots`](https://docs.rs/rustls-platform-verifier/latest/rustls_platform_verifier/struct.Verifier.html#method.new_with_extra_roots); however, it doesn't exist on e.g. macOS and Windows, so one would either need a platform-conditional API or at least new error codes for these cases. WDYT?

Reply:
There's an open issue (https://github.com/rustls/rustls-platform-verifier/issues/58) on rustls-platform-verifier to add support for extra roots for the other platforms.

I think we should keep this simple and avoid implementing it in -ffi until the feature is available more broadly. There's not very much platform-conditional code in this repo and it'd be great to avoid adding more if we can.